android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -1588,6 +1610,84 @@ protected void onSizeChanged(int w, int h, int ow, int oh) {
           )
         );
       }
+    }   
+
+    @Override
+    public ActionMode startActionMode(ActionMode.Callback callback, int type) {
+      if(menuItems == null || menuItems.size() == 0){
+        return super.startActionMode(callback, type);
+      }
+      return super.startActionMode(new ActionMode.Callback2() {
+        @Override
+        public boolean onCreateActionMode(ActionMode mode, Menu menu) {
+          for(int i = 0; i < menuItems.size(); i++ ){
+            menu.add(Menu.NONE, i, i, ((HashMap<String, String>)menuItems.get(i)).get("label"));
+          }
+          return true;
+        }
+
+        @Override
+        public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
+          WritableMap wMap = Arguments.createMap();
+          HashMap<String, String> menuItemMap = (HashMap<String, String>)menuItems.get(item.getItemId());
+          wMap.putString("label", menuItemMap.get("label"));
+          wMap.putString("key", menuItemMap.get("key")); 
+          wMap.putString("selectedText", RNCWebView.this.selectionText); 
+          ReactContext reactContext = (ReactContext)RNCWebView.this.getContext();
+          reactContext
+            .getJSModule(RCTEventEmitter.class)
+            .receiveEvent(getId(), "customMenuSelection", wMap);
+          mode.finish();
+          return true;
+        }
+
+        @Override
+        public void onDestroyActionMode(ActionMode mode) {
+          mode = null;
+        }
+
+        @Override
+        public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
+          return true;
+        }
+
+        @Override
+        public void onGetContentRect (ActionMode mode, 
+                View view, 
+                Rect outRect){
+
+          String js = "(function(){ let wr = " + RNCWebView.this.getWidth() + "/window.innerWidth; let hr = " + RNCWebView.this.getHeight() +"/window.innerHeight; let rect = window.getSelection()?.getRangeAt(0)?.getBoundingClientRect(); return JSON.stringify({x: (rect?.x * wr), y: (rect?.y * hr), width: (rect?.width * wr), height: (rect?.height * hr), selectionText: window.getSelection().toString()?.replace(/\"/g, '%%22')?.replace(/\\\\/g, '%%5C')})})()";
```

Comment:
Since you don't replace the variables later, you can use `const` rather than `let`.